scrapy_zyte_api/page_inputs.py

```diff
@@ -0,0 +1,55 @@
+from typing import Annotated, List, Optional, TypedDict
+
+import attrs
+
+
+class _Selector(TypedDict, total=False):
+    type: str
+    value: str
+    state: Optional[str]
+
+
+class _Action(TypedDict, total=False):
+    action: str
+    url: Optional[str]
+    urlPattern: Optional[str]
+    urlMatchingOptions: Optional[str]
+    selector: Optional[_Selector]
+    onError: Optional[str]
+    delay: Optional[float]
+    timeout: Optional[float]
+    id: Optional[str]
+    keyword: Optional[str]
+    key: Optional[str]
+    source: Optional[str]
+    options: Optional[dict]
```

Comment:
Wouldn’t it be better to have an approach that does not require defining action top-level fields here?

Reply:
This section is not required; the code would work fine without it as-is. If there are other top-level fields, I think it also should work as-is. I added it to have some type annotations for the users, via TypedDict. Would you prefer to have it not type checked?

Reply:
I was not familiar with `TypedDict`, nice.